### PR TITLE
vscode: use absolute predetermined path instead of heuristic

### LINKED ISSUE #59725: VSCode: No such file or directory with User Env
## Issue description

I have the latest version of VSCode installed (`nixpkgs` channel). I tried running `code` in my shells (zsh, bash, sh) and all of them failed with this error:

```
$ ~/ code
-f: No such file or directory
dirname: missing operand
Try 'dirname --help' for more information.
/nix/store/nhfb7dsg4h010nqh7a64cd3v96yvjp9v-vscode-1.33.0/bin/.code-wrapped: line 35: /../code: No such file or directory
```

My `dirname` version: `dirname (GNU coreutils) 8.29`.

**However**, `nix-shell -p vscode --run "code"` runs fine. What could possibly be interfering?

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.107, NixOS, 18.09.2367.15f08994517 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(max): `"nixpkgs-19.09pre174594.0c0954781e2"`
 - channels(root): `"nixos-18.09.2505.46d3867a08a, nixos-hardware, unstable-19.03pre166740.422b6bd489a"`
 - nixpkgs: `/home/max/.nix-defexpr/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by Synthetica9**
Fixes #59725

###### Motivation for this change

#59725

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/editors/vscode/default.nix [hunk 1]
```diff
@@ -108,6 +108,9 @@ in
 
         mkdir -p $out/share/pixmaps
         cp $out/lib/vscode/resources/app/resources/linux/code.png $out/share/pixmaps/code.png
+
+        # Override the previously determined VSCODE_PATH with the one we know to be correct
+        sed -i "/ELECTRON=/iVSCODE_PATH='$out/lib/vscode'" $out/bin/code
```
Comment (Mic92) [inline pkgs/applications/editors/vscode/default.nix]:
```suggestion
        sed -i "/ELECTRON=/iVSCODE_PATH='$out/lib/vscode'" $out/bin/code
        grep -q "VSCODE_PATH='$out/lib/vscode'" $out/bin/code # check if sed succeeded
```

Comment (Synthetica9) [reply pkgs/applications/editors/vscode/default.nix]: Doesn't sed fail if it doesn't succeed?

Comment (Mic92) [reply pkgs/applications/editors/vscode/default.nix]: No it does not.

**Comment by Mic92** [comment]
Does this need a backport?

**Comment by worldofpeace** [comment]
Most changes to vscode should be backported

File: pkgs/applications/editors/vscode/default.nix [hunk 2]
```diff
@@ -108,6 +108,10 @@ in
...
```
Comment (eadwu) [inline pkgs/applications/editors/vscode/default.nix]: `$out/bin/code` should be `$out/bin/${executableName}`

Comment (eadwu) [reply pkgs/applications/editors/vscode/default.nix]: See #59798.

**Pull request merged**